File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hash_set={}
        for i in range(0,len(nums)):
            val1=nums[i]
            val2=target-val1
            if val2 not in hash_set:
                hash_set[val1]=i
            else:
                return hash_set[val2],i

        


        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> ps= new HashMap<>();
        int[] ans= new int[2];
        for(int i=0;i<nums.length;i++){
            int element2=target-nums[i];
            if(ps.containsKey(element2)){
                ans[0]=ps.get(element2);
                ans[1]=i;
                return ans;
            }else{
                ps.put(nums[i],i);
            }
        }
        return ans;
    }
}